lib/smack/SmackOptions.cpp

```diff
@@ -14,8 +14,20 @@ const llvm::cl::list<std::string> SmackOptions::EntryPoints(
   llvm::cl::value_desc("PROCS")
 );
 
-const llvm::cl::opt<bool> SmackOptions::Warnings(
-  "warnings", llvm::cl::desc("Enable warnings.")
+const llvm::cl::opt<SmackWarnings::WarningLevel> SmackOptions::WarningLevel(
+  "warn-type",
+  llvm::cl::desc("Enable certain type of warning messages."),
+  llvm::cl::values(
+    clEnumValN(SmackWarnings::WarningLevel::Silent,
+      "silent", "No warning messages"),
+    clEnumValN(SmackWarnings::WarningLevel::Unsound,
+      "unsound", "Enable warnings about unsoundness"),
+    clEnumValN(SmackWarnings::WarningLevel::Info,
+      "info", "Enable warnings about unsoundness and debug information"))
```

Comment:
It is strange to call this `debug information`. In our recent discussion, you said that we are still using `DEBUG` macro for printing debug information. So this should be named different. What does `info` enable?

Reply:
It enables memcpy/memset length warning and collapsing DSA nodes. I'm not sure what is a good name for these.

Reply:
Maybe `translation information`. So `Enable warnings about unsoundness and translation information`. You can fix this below as well, and then maybe squash commits. Thanks!